### PR TITLE
Sqlitedb API implementation

### LINKED ISSUE #16: Sqlite API
Implement an AerovalDB implementation which uses sqlite to store data. For now it should store data in json blobs equivalent to what is read in the json api.

### LINKED ISSUE #52: AerovalDB should support more than one db-format
AerovalDB should at least support 2 formats natively to prove that it supports more than one format. Besides json-files, other formats should include

- a relational dbms, e.g. sqlite (see #16 )
- a key-value store, e.g. rocksdb

### PR DESCRIPTION
**Pull request by thorbjoernl**
## Change Summary

Provides an initial implementation for storing json blobs in an sqlite database.

image

## Related issue number

- closes #16, 

- closes #52

## Checklist

* [x] Start with a draft-PR
* [x] The PR title is a good summary of the changes
* [x] PR is set to AeroTools and a tentative milestone
* [ ] Documentation reflects the changes where applicable
* [x] Tests for the changes exist where applicable
* [x] Tests pass locally
* [x] Tests pass on CI
* [x] At least 1 reviewer is selected
* [x] Make PR ready to review

### CODE REVIEW

File: src/aerovaldb/utils/string_mapper/mapper.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
+import logging
+from typing import Awaitable, Callable, Mapping
+from abc import ABC
+from packaging.version import Version
+
+logger = logging.getLogger()
+
```
Comment (heikoklein) [inline src/aerovaldb/utils/string_mapper/mapper.py:7]: Shouldn't that be 
```python
logger = logging.getLogger(__name__)
```

File: src/aerovaldb/jsondb/jsonfiledb.py [hunk 2]
```diff
@@ -241,9 +148,9 @@ async def _get_version(self, project: str, experiment: str) -> Version:
                 version = Version("0.0.1")
             finally:
                 return version
-        except simplejson.errors.JSONDecodeError:
-            # Work around for https://github.com/metno/aerovaldb/issues/28
-            return Version("0.14.0")
+        # except simplejson.JSONDecodeError:
+        #    # Work around for https://github.com/metno/aerovaldb/issues/28
+        #    return Version("0.14.0")
```
Comment (heikoklein) [inline src/aerovaldb/jsondb/jsonfiledb.py:153]: Dead code?

**Review by heikoklein — APPROVED** [review]
Congratulations, having two independent implementations is an important step for aerovaldb.

With this step you also refactored much of jsondb code and testing, which was expected, but is also a bit overwhelming when reading the PR. I would have prefered several smaller PRs, e.g. about the refactoring of jsondb, changes introduced in testing and then only sqlitedb. Now, it is impossilbe to see for me where a change is coming from and where it might eventually break functionality. But this just as a note for the future.

What is completely missing is the documention on 'How to implement add an additional aerovaldb implementation', similar to https://pyaro.readthedocs.io/en/latest/how-to-add-new-reader.html . With the addition of sqlitedb you made a lot of decisions about that, and that should be documented. Since the documentation isn't strictly necessary for this PR, I approve.

**Comment by thorbjoernl** [comment]
Thanks for reviewing :) I'll try to keep changes smaller and more manageable in the future.

I've created an issue for the documentation here for future reference: https://github.com/metno/aerovaldb/issues/80

**Pull request merged**